sycl/include/sycl/builtins.hpp

```diff
@@ -775,6 +776,92 @@ detail::enable_if_t<detail::is_svgenfloat<T>::value, T> trunc(T x) __NOEXC {
   return __sycl_std::__invoke_trunc<T>(x);
 }
 
+// other marray math functions
+
+// TODO: can be optimized in the way marray math functions above are optimized
+// (usage of vec<T, 2>)
+#define __SYCL_MARRAY_MATH_FUNCTION_W_GENPTR_ARG_OVERLOAD_IMPL(NAME, ARGPTR,   \
+                                                               ...)            \
+  marray<T, N> res;                                                            \
+  for (int j = 0; j < N; j++) {                                                \
+    multi_ptr<detail::marray_element_t<detail::remove_pointer_t<T2>>,          \
+              T2::address_space>                                               \
+        ptr = nullptr;                                                         \
+    res[j] = NAME(__VA_ARGS__, ptr);                                           \
+    (*ARGPTR)[j] = *ptr;                                                       \
```

Comment:
This doesn't look right. Dereferencing `nullptr` is not a good idea. Or am I missing something?

I though that you should pass an address there where to put the result, i.e. it should be `multi_ptr(&ARGPTR[j])` or something like that passed to the built-in instead of `ptr`

Reply:
Agree, and while the failure didn't show up for these tests specifically on GPU device, there is always a segfault on CPU device.

It looks like it is impossible to call `multi_ptr(&ARGPTR[j])` at least for the current SYCL 2020 spec - there is no ctor for that, but if it was, it would have definitely fixed this.